src/Components/Workshop/WorkshopBar.js

```diff
@@ -24,15 +24,34 @@ class WorkshopBar extends React.Component {
   }
 
   render() {
+    //extract the day, month, year from the date Object
+    let date = null;
+    let month = null;
+    let year = null;
+    let day = null;
+    if (this.props.data.Date != null) {
+      date = this.props.data.Date.getDate();
+      
+      month = this.props.data.Date.getMonth();
+      let monthsOfYear = ['January','February','March','April','May','June','July','August','September','October','November','December'];
+      month = monthsOfYear[month];
+
+      year = this.props.data.Date.getFullYear();
+      
+      day = this.props.data.Date.getDay();
+      let daysOfWeek = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+      day = daysOfWeek[day];
+    }
```

Comment:
```javascript
  //extract the day, month, year from the date Object
    let date = null;
    let month = null;
    let year = null;
    let day = null;
    if (this.props.data.Date != null) {
      date = this.props.data.Date.getDate();

      month = this.props.data.Date.getMonth();
      let monthsOfYear = ['January','February','March','April','May','June','July','August','September','October','November','December'];
      month = monthsOfYear[month];

      year = this.props.data.Date.getFullYear();

      day = this.props.data.Date.getDay();
      let daysOfWeek = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
      day = daysOfWeek[day];
    }
```
Make this code a function. A lot of logic going on, would look neater in my opinion rather than having it inside the `render()`.

Reply:
The reason I've scoped it to be inside render is because only what the end user sees is modified. To extract all the logic out to a sepearate method would also mean increasing the scope of the variables and there is no requirement to do that. Its purely a visual change that happens right before the date gets rendered to the screen